pandas/core/indexes/base.py

```diff
@@ -4702,6 +4702,13 @@ def dropna(self, how='any'):
     def _evaluate_with_timedelta_like(self, other, op):
         # Timedelta knows how to operate with np.array, so dispatch to that
         # operation and then wrap the results
+        if self._is_numeric_dtype and op.__name__ in ['add', 'sub',
+                                                      'radd', 'rsub']:
+            raise TypeError("Operation {opname} between {cls} and {other} "
+                            "is invalid".format(opname=op.__name__,
+                                                cls=type(self).__name__,
```

Comment:
Actually, I think we should use here the name of the dtype instead of the class, as this will bubble up for Series as well

```
In [3]: ser + tdnat
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-4ad1d16388f8> in <module>()
----> 1 ser + tdnat

~/scipy/pandas/pandas/core/ops.py in wrapper(left, right)
   1419             #  that may incorrectly raise TypeError when we
   1420             #  should get NullFrequencyError
-> 1421             result = op(pd.Index(left), right)
   1422             return construct_result(left, result,
   1423                                     index=left.index, name=res_name,

~/scipy/pandas/pandas/core/indexes/base.py in index_arithmetic_method(self, other)
    137         # handle time-based others
    138         if isinstance(other, (ABCDateOffset, np.timedelta64, timedelta)):
--> 139             return self._evaluate_with_timedelta_like(other, op)
    140         elif isinstance(other, (datetime, np.datetime64)):
    141             return self._evaluate_with_datetime_like(other, op)

~/scipy/pandas/pandas/core/indexes/base.py in _evaluate_with_timedelta_like(self, other, op)
   4708                             "is invalid".format(opname=op.__name__,
   4709                                                 cls=type(self).__name__,
-> 4710                                                 other=type(other).__name__))
   4711 
   4712         other = Timedelta(other)

TypeError: Operation add between Int64Index and timedelta64 is invalid
```

Reply:
will update in follow-up PR that addresses the bug above.